iterators.tex

```diff
@@ -4287,6 +4394,173 @@
+\tcode{last}. If \tcode{S} is a literal type, this constructor
+shall be a \tcode{constexpr} constructor.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{move_iterator}!constructor}%
+\begin{itemdecl}
+explicit move_sentinel(S s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Constructs a \tcode{move_sentinel}, initializing
+\tcode{last} with \tcode{s}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{move_sentinel}!constructor}%
+\begin{itemdecl}
+move_sentinel(const move_sentinel<ConvertibleTo<S>>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Constructs a \tcode{move_sentinel}, initializing
+\tcode{last} with \tcode{s.last}.
+\end{itemdescr}
+
+\rSec4[move.sent.op=]{\tcode{move_sentinel::operator=}}
+
+\indexlibrary{\idxcode{operator=}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator=}}%
+\begin{itemdecl}
+move_sentinel& operator=(const move_sentinel<ConvertibleTo<S>>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Assigns \tcode{s.last} to \tcode{last}.
+\end{itemdescr}
+
+\rSec4[move.sent.op.comp]{\tcode{move_sentinel} comparisons}
+
+\indexlibrary{\idxcode{operator==}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator==}}%
+\begin{itemdecl}
+template <class I, Sentinel<I> S>
+  bool operator==(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+template <class I, Sentinel<I> S>
+  bool operator==(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{i.current == s.last}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{operator"!=}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator"!=}}%
+\begin{itemdecl}
+template <class I, Sentinel<I> S>
+  bool operator!=(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+template <class I, Sentinel<I> S>
+  bool operator!=(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{!(i == s)}.
+\end{itemdescr}
+
+\rSec4[move.sent.nonmember]{\tcode{move_sentinel} non-member functions}
+
+\indexlibrary{\idxcode{operator-}!\idxcode{move_sentinel}}%
+\indexlibrary{\idxcode{move_sentinel}!\idxcode{operator-}}%
+\begin{itemdecl}
+template <class I, SizedSentinel<I> S>
+  difference_type_t<I> operator-(
+    const move_sentinel<S>& s, const move_iterator<I>& i);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{s.last - i.current}.
+\end{itemdescr}
+
+\begin{itemdecl}
+template <class I, SizedSentinel<I> S>
+  difference_type_t<I> operator-(
+    const move_iterator<I>& i, const move_sentinel<S>& s);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects Equivalent to \tcode{i.current - s.last}.
+\end{itemdescr}
+
+\indexlibrary{\idxcode{make_move_sentinel}}%
+\begin{itemdecl}
+template <Semiregular S>
+  move_sentinel<S> make_move_sentinel(S s);
```

Comment:
>  I suppose you could argue that since move_iterator is Input, it doesn't matter.

I might argue that it's not a pessimization since `move_iterator` is always input, but there's clearly a loss of information. This is almost the identical discussion we had about common iterators months ago in ericniebler/stl2#37; separately converting an iterator and sentinel is generally less effective than converting ranges to ranges due to lack of information about the relationship between the iterator and sentinel. The argument there for `make_bounded_range` applies here as well: `make_move_range` would be more effective, but encroaches on `view` territory.

I'd prefer to leave this as-is and solve the issue with `view::move` rather than specifying ad-hoc `make_foo_range` functions that will be redundant once views are added.